src/components/dialog-test.js

```diff
@@ -411,6 +411,86 @@ describes.realWin('Dialog', (env) => {
       expect(styleDuringInit).to.equal('display: none !important;');
     });
 
+    describe('fade background', () => {
+      it('should not register click event by default', async () => {
+        const old = Dialog.prototype.onGrayPaneClick_;
+        let wasClicked = false;
+        Dialog.prototype.onGrayPaneClick_ = function () {
+          wasClicked = true;
+        };
+
+        dialog = new Dialog(globalDoc, {});
+        const el = dialog.graypane_.getElement();
+        const openedDialog = await dialog.open(NO_ANIMATE);
+        await openedDialog.openView(view);
+
+        el.click();
+
+        expect(wasClicked).to.be.false;
+
+        Dialog.prototype.onGrayPaneClick_ = old;
+      });
+
+      it('should register click event', async () => {
+        const old = Dialog.prototype.onGrayPaneClick_;
+        let wasClicked = false;
+        Dialog.prototype.onGrayPaneClick_ = function () {
+          wasClicked = true;
+        };
+
+        dialog = new Dialog(globalDoc, {}, {}, {isClosable: false});
+        const el = dialog.graypane_.getElement();
+        const openedDialog = await dialog.open(NO_ANIMATE);
+        await openedDialog.openView(view);
+
+        el.click();
+
+        expect(wasClicked).to.be.true;
+
+        Dialog.prototype.onGrayPaneClick_ = old;
+      });
```

Comment:
Whether a click event is registered is an implementation detail. The important behavior to verify is whether clicks are blocked/the dialog is closed.

Reply:
Fixed